src/js/parsers/rssparser.ts

```diff
@@ -31,17 +38,25 @@ export default function parseRss(dat) {
 }
 
 // Translate RSS item to playlist item.
-function parseItem(obj) {
-    const item = {};
+function parseItem(obj: PageNode): PlaylistItemType {
+    const item: {
+        file?: string;
+        title?: string;
+        mediaid?: string;
+        date?: string;
+        description?: string;
+        link?: string;
+        tags?: string;
+    } = {};
     for (let i = 0; i < obj.childNodes.length; i++) {
-        const node = obj.childNodes[i];
+        const node: PageNode = obj.childNodes[i];
         const name = localName(node);
         if (!name) {
             continue;
         }
         switch (name.toLowerCase()) {
             case 'enclosure':
-                item.file = xmlAttribute(node, 'url');
+                item.file = xmlAttribute(node as Element, 'url');
```

Comment:
@zetagame can't figure out a way around this one, i tried redefining `const n:Element = node;` above this and it errors out. reasoning is `obj.childNodes` returns a `ChildNode` and since the `PageNode` type is a union type of `(Element | ChildNode)` that decision is made higher and it sticks to this being explicitly a `ChildNode` - I wonder if there's a built in type that works for this however I tried a few to no avail so I'm willing to commit this as is. Thoughts?

Reply:
Good catch! I have a PR up now for a proposed solution.